fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/JoinedRecordTypeTest.java

```diff
@@ -0,0 +1,208 @@
+                    syntheticRecordTypeKey.getItems().get(0),
+                    simpleRecord.getPrimaryKey().getItems(),
+                    otherRecord.getPrimaryKey().getItems());
+
+            context.commit();
+        }
+
+        try (FDBRecordContext context = openContext()) {
+            createOrOpenRecordStore(context, baseMetaData(addJoinedType()));
+
+            recordStore.deleteRecord(otherRecord.getPrimaryKey());
+
+            context.commit();
+        }
+
+        try (FDBRecordContext context = openContext()) {
+            createOrOpenRecordStore(context, baseMetaData(addJoinedType()));
+            // Default policy (ERROR) should fail
+            final ExecutionException exception = assertThrows(ExecutionException.class, () -> recordStore.loadSyntheticRecord(joinedPrimaryKey).get());
+            assertEquals(RecordDoesNotExistException.class, exception.getCause().getClass());
+            // RETURN policy returns the shell of the synthetic record with no constituents
+            final FDBSyntheticRecord syntheticRecord = recordStore.loadSyntheticRecord(joinedPrimaryKey, IndexOrphanBehavior.RETURN).get();
+            assertEquals(0, syntheticRecord.getConstituents().size());
+            // SKIP policy returns null in case of missing constituents
+            assertNull(recordStore.loadSyntheticRecord(joinedPrimaryKey, IndexOrphanBehavior.SKIP).get());
+
+            context.commit();
+        }
+    }
+
+    @Test
+    void loadSyntheticRecordFailsMissingAllConstituents() throws ExecutionException, InterruptedException {
+        final Tuple joinedPrimaryKey;
+        final FDBStoredRecord<Message> simpleRecord;
+        final FDBStoredRecord<Message> otherRecord;
+        try (FDBRecordContext context = openContext()) {
+            createOrOpenRecordStore(context, baseMetaData(addJoinedType()));
+
+            Tuple syntheticRecordTypeKey = recordStore.getRecordMetaData()
+                    .getSyntheticRecordType(JOINED_RECORD_NAME)
+                    .getRecordTypeKeyTuple();
+            simpleRecord = recordStore.saveRecord(createSimpleRecord(100, 10));
+            otherRecord = recordStore.saveRecord(createOtherRecord(101, 11));
+            joinedPrimaryKey = Tuple.from(
+                    syntheticRecordTypeKey.getItems().get(0),
+                    simpleRecord.getPrimaryKey().getItems(),
+                    otherRecord.getPrimaryKey().getItems());
+
+            context.commit();
+        }
+
+        try (FDBRecordContext context = openContext()) {
+            createOrOpenRecordStore(context, baseMetaData(addJoinedType()));
+
+            recordStore.deleteRecord(simpleRecord.getPrimaryKey());
+            recordStore.deleteRecord(otherRecord.getPrimaryKey());
+
+            context.commit();
+        }
+
+        try (FDBRecordContext context = openContext()) {
+            createOrOpenRecordStore(context, baseMetaData(addJoinedType()));
+            // Default policy (ERROR) should fail
```

Comment:
It's probably worth calling with explicitly specifying `ERROR`

Reply:
Done.

Reply:
Sorry, I meant, in addition to the default.